.github/workflows/release.yaml

```diff
@@ -0,0 +1,79 @@
+name: Publish new release
+
+on:
+  push:
+    tags:
+      - "v*"
+
+jobs:
+  create_release:
+    name: Create Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@master
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: Release ${{ github.ref }}
+          body: |
+            Release ${{ github.ref }}
+          draft: false
+          prerelease: false
+
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    needs: create_release
+    steps:
+      - name: Set up Go 1.17
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.17
+        id: go
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v1
+      - name: Restore cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Get dependencies
+        run: |
+          go mod download
+      - name: Build analyzer binary
+        run: |
+          mkdir -p releases/ build/
+          for platform in linux darwin windows
+          do
+            for arch in arm64 amd64
```

Comment:
Just an idea for the future, but would be nice to get this into a build matrix instead for building all versions in parallel 
https://docs.github.com/en/actions/using-workflows/advanced-workflow-features#using-a-build-matrix

Reply:
Matrix creates new job for every platform - which takes more time. With pure Go project there is no point to use them - Go perfectly cross-compiles to every platform. 
Using matrixes make sense when project uses CGO - so cross-compiling is not possible.